tests/common/common.py

```diff
@@ -172,6 +172,57 @@ def test_monitored_resource(self):
             self.assertTrue(found_resource.labels[label],
                 f'resource.labels[{label}] is not set')
 
+    def test_request_log(self):
+        if self.language not in ["nodejs"]:
+            # TODO: other languages to also support this test
+            return True
+        log_text = f"{inspect.currentframe().f_code.co_name}"
+        log_list = self.trigger_and_retrieve(log_text, "requestlog")
+        found_request = log_list[-1].http_request
+        # TODO(nicolezhu): remove hasattr check later.
+        if hasattr(self, 'request_props'):
+            for prop in self.request_props:
+                self.assertTrue(found_request[prop],
+                f'httpRequest[{prop}] is not set')
+
+    def test_stdout_log(self):
```

Comment:
Should this test really be in the "common" class? Common is for tests that run in all environments, but standard out isn't captured in all environments (eg compute engine).

It may make more sense to create a separate superclass just shared by stdout environments? But for Python, I never test for standard out explicitly, I just trigger logs in each environment and make sure they all work consistently, no matter which handler is used

Reply:
oke i ll move it to a different file, and good point on filtering out GCE.